### PR TITLE
added candidate service and migrated getcandidatebyid need a pr on this

### PR DESCRIPTION
**Pull request by airinnlionhart**
I am just looking at my changes for this PR. Just let me know where I am not falling in. Best practices do not have to go too much in-depth.

### CODE REVIEW

File: Candidate/Services/CandidateService.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                        }
+                    }
+                    return candidate;
+                }
+            }
+            catch (Exception ex)
+            {
+                // Handle or log the exception
+                Console.WriteLine("Error unable to retrieve candidate from database: " + ex.Message);
+                return null;
```
Comment (airinnlionhart) [inline Candidate/Services/CandidateService.cs:62]: throw

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,5 @@
 Candidate/appsettings.json 
 Candidate/appsettings.local.json
+bin
+obj
+.vs
```
Comment (unaware1224) [inline .gitignore]: Note, you'll also need to now delete all the files in these directories to clean up the repo. 
```suggestion
.vs/
```

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,5 @@
 Candidate/appsettings.json 
 Candidate/appsettings.local.json
+bin
+obj
```
Comment (unaware1224) [inline .gitignore]:
```suggestion
[Oo]bj/
```

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,5 @@
 Candidate/appsettings.json 
 Candidate/appsettings.local.json
+bin
```
Comment (unaware1224) [inline .gitignore]:
```suggestion
[Bb]in/
```

File: Candidate/Controllers/CandidateController.cs [hunk 5]
```diff
@@ -173,49 +176,21 @@ public IActionResult DeleteCandidate(int id)
                 return StatusCode(500, $"An error occurred: {ex.Message}");
             }
         }
-        [HttpGet("GetCandidate/{id}")]
-        public IActionResult GetCandidate(int id)
+        [HttpGet("GetCandidateById/{id}")]
```
Comment (unaware1224) [inline Candidate/Controllers/CandidateController.cs]:
```suggestion
        [HttpGet("{id}")]
```

File: Candidate/Controllers/CandidateController.cs [hunk 6]
```diff
@@ -173,49 +176,21 @@ public IActionResult DeleteCandidate(int id)
...
+        public IActionResult GetCandidateById(int id)
```
Comment (unaware1224) [inline Candidate/Controllers/CandidateController.cs]: At some point, consider switching to async for best server performance (handle more load)

File: Candidate/Services/CandidateService.cs [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            _configuration = configuration;
+        }
+
+        public Candidate.Models.Candidate GetCandidate(int id)
+        {
+            try
+            {
+                Candidate.Models.Candidate candidate = null;
+
+                using (SqlConnection connection = new SqlConnection(_configuration.GetConnectionString("DefaultConnection")))
```
Comment (unaware1224) [inline Candidate/Services/CandidateService.cs]: Consider creating a central place to get the SqlConnection ... this way if the connection string name changes, its a single change, not a hundred.

File: Candidate/Services/CandidateService.cs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+                {
+                    connection.Open();
+
+                    string selectDataSql = "SELECT * FROM candidate WHERE id = @id;";
+
+                    using (SqlCommand command = new SqlCommand(selectDataSql, connection))
+                    {
+                        command.Parameters.AddWithValue("@id", id);
```
Comment (unaware1224) [inline Candidate/Services/CandidateService.cs]: Kudos for using parameters

File: Candidate/Services/CandidateService.cs (hunk 1 above)
Comment (unaware1224) [inline Candidate/Services/CandidateService.cs:62]:
```suggestion
                throw;
```

File: Candidate/Services/CandidateService.cs [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+                            }
+                        }
+                    }
+                    return candidate;
+                }
+            }
+            catch (Exception ex)
+            {
+                // Handle or log the exception
+                Console.WriteLine("Error unable to retrieve candidate from database: " + ex.Message);
```
Comment (unaware1224) [inline Candidate/Services/CandidateService.cs:61]: Consider using the .NET logging library, that works with any logging provider.

File: Candidate/Services/CandidateService.cs (hunk 7 above)
Comment (airinnlionhart) [reply Candidate/Services/CandidateService.cs]: Great Idea

Comment (airinnlionhart) [reply Candidate/Services/CandidateService.cs]: Added

File: Candidate/Services/CandidateService.cs (hunk 1 above)
Comment (airinnlionhart) [reply Candidate/Services/CandidateService.cs:62]: addded

File: Candidate/Controllers/CandidateController.cs (hunk 6 above)
Comment (airinnlionhart) [reply Candidate/Controllers/CandidateController.cs]: Added

**Pull request merged**